File: benchmark_array_sum.c
#include <stdio.h>

// Array summation benchmark - C version
int array_sum() {
    int sum = 0;
    for (int i = 0; i < 1000000; i++) {
        sum += i;
    }
    return sum;
}

int main() {
    int result = array_sum();
    printf("Array sum result: %d\n", result);
    return 0;
}

File: benchmark_array_sum.rs
// Array summation benchmark - Rust version
fn array_sum() -> i32 {
    let mut sum = 0;
    for i in 0..1000000 {
        sum += i;
    }
    sum
}

fn main() {
    let result = array_sum();
    println!("Array sum result: {}", result);
}